### PR TITLE
[FEATURE][CU-865ctbjwt] Patient Onboarding

### PR DESCRIPTION
**Pull request by cgarcia-lightit**
Feature/cu 865ctbjwt patient onboarding

# Light It Pull Request Template ⚡⚡⚡

_Resolve_: [CU-865ctbjwt](https://app.clickup.com/t/865ctbjwt)

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: apps/eo_web/src/api/useElixirApi.ts [hunk 1]
```diff
@@ -108,6 +109,27 @@ export const useElixirApi = () => {
     );
   };
 
+  const getSubmissionByIdV2 = async (submissionId: string) => {
+    return await api.get<CancerForm>(
+      `${API_URL}/v2/submission/cancer?submission_id=${submissionId}`,
+      authHeader,
+    );
+  };
```
Comment (UrielRadzyminski1) [inline apps/eo_web/src/api/useElixirApi.ts:117]: Comment here to rememeber to remove this

File: apps/eo_web/src/components/Header.tsx [hunk 2]
```diff
@@ -35,6 +35,9 @@ export const Header = () => {
         src="https://assets-global.website-files.com/641990da28209a736d8d7c6a/641990da28209a61b68d7cc2_eo-logo%201.svg"
         alt="Leters EO"
         className="h-11 w-20"
+        onClick={() => {
+          window.location.href = "/";
```
Comment (UrielRadzyminski1) [inline apps/eo_web/src/components/Header.tsx:39]: Why not use react router push?

Comment (UrielRadzyminski1) [reply apps/eo_web/src/components/Header.tsx:39]: If you want this, use an <a> tag

File: apps/eo_web/src/screens/Cancer/CancerForm.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+
+const CANCER_PROFILE_ID = window.data.CANCER_PROFILING || 232054030821037;
+
+export const CancerForm = () => {
+  const location = useLocation();
+
+  const { submission_id } = location.state
+    ? (location.state as { submission_id: string })
+    : { submission_id: "" };
```
Comment (UrielRadzyminski1) [inline apps/eo_web/src/screens/Cancer/CancerForm.tsx:21]: Why this?

File: apps/eo_web/src/screens/Cancer/CancerThankYou.tsx [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+  const navigate = useNavigate();
+
+  if (!submission_id) {
+    navigate(ROUTES.cancerProfile);
+  }
+
+  const { getSubmissionByIdV2, postCancerFormSubmission } = useElixirApi();
+  const { data } = useQuery({
+    queryFn: () => getSubmissionByIdV2(submission_id),
+    queryKey: ["cancerLastSubmission"],
```
Comment (UrielRadzyminski1) [inline apps/eo_web/src/screens/Cancer/CancerThankYou.tsx:58]: Add submission_id to queryKey

File: apps/eo_web/src/screens/Cancer/CancerThankYou.tsx [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+    if (typeof value === "string") {
+      return (
+        Object.keys(enumObj).find((key) => enumObj[key] === value) || value
+      );
+    } else {
+      return value.map((val) => getKeyByValue(enumObj, val) as string);
+    }
+  }
+
+  const makeRequest = (e: CancerFormInterface) => {
```
Comment (UrielRadzyminski1) [inline apps/eo_web/src/screens/Cancer/CancerThankYou.tsx:97]: Move this function somewhere else?

File: apps/eo_web/src/screens/Cancer/CancerThankYou.tsx [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+        if (result.response?.status !== 200) {
+          toast.error("Something went wrong");
+        }
+      } else {
+        toast.error("Something went wrong");
+      }
+    },
+  });
+
+  function getKeyByValue(
```
Comment (UrielRadzyminski1) [inline apps/eo_web/src/screens/Cancer/CancerThankYou.tsx:84]: Move this function somewhere else?

File: apps/eo_web/src/screens/Cancer/UserCancerProfile.tsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
+import { useEffect } from "react";
+
+import { jotformScript } from "~/helpers/jotform_script";
+import { LayoutDefault } from "~/layouts";
+
+
+
+
+
```
Comment (UrielRadzyminski1) [inline apps/eo_web/src/screens/Cancer/UserCancerProfile.tsx:9]: Why this space?

File: apps/eo_web/src/screens/Cancer/UserCancerVerification.tsx [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+
+import { Loading, Modal, Typography } from "@eo/ui";
+
+import { useElixirApi } from "~/api/useElixirApi";
+import { LayoutDefault } from "~/layouts";
+import { ROUTES } from "~/router";
+
+
+
+
```
Comment (UrielRadzyminski1) [inline apps/eo_web/src/screens/Cancer/UserCancerVerification.tsx:13]: Why space?

File: apps/eo_web/src/screens/Cancer/UserCancerVerification.tsx [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+  const submissionId = searchParams.get("submission_id") || "";
+  if (!email || !submissionId) {
+    navigate(ROUTES.cancerProfile);
+  }
+
+  const [openModal, setOpenModal] = useState(false);
+  const [closeModal, setCloseModal] = useState(false);
+  const { data: response, isLoading } = useQuery({
+    queryFn: () => eligibleEmail(email),
+    queryKey: ["eligibleEmail"],
```
Comment (UrielRadzyminski1) [inline apps/eo_web/src/screens/Cancer/UserCancerVerification.tsx:31]: Add email to queryKey

File: apps/eo_web/src/components/Header.tsx (hunk 2 above)
Comment (UrielRadzyminski1) [reply apps/eo_web/src/components/Header.tsx:39]: an <a> tag

File: apps/eo_web/src/screens/Cancer/CancerForm.tsx [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+  const last = searchParams.get("last");
+  const dob = searchParams.get("dob");
+  const email = searchParams.get("email");
+  const caregiver = searchParams.get("caregiver");
+  const submission_id = searchParams.get("submission_id");
+  const [day, month, year] = dob?.split("-") || [];
+
+  const navigate = useNavigate();
+
+  if (!submission_id) {
```
Comment (UrielRadzyminski1) [inline apps/eo_web/src/screens/Cancer/CancerForm.tsx:27]: Porque necesitas el submission id aca?

Comment (cgarcia-lightit) [reply apps/eo_web/src/screens/Cancer/CancerForm.tsx:27]: porque si no viene, quiere decir que esta accediendo directamente a la url. y no puedo empezar el flujo desde el formulario de cancer, asi que lo devuelvo para que complete los datos personales. De lo contrario el form quedaria incompleto.

**Pull request merged**